Os/Stub/test/ut/StubFileTests.cpp

```diff
@@ -0,0 +1,215 @@
+// ======================================================================
+// \title Os/Stub/test/ut/StubFileTests.cpp
+// \brief tests using stub implementation for Os::File interface testing
+// ======================================================================
+#include <gtest/gtest.h>
+#include "Os/File.hpp"
+#include "Os/Models/Models.hpp"
+#include "Os/test/ut/file/CommonTests.hpp"
+#include "Os/test/ut/file/RulesHeaders.hpp"
+#include "Os/Stub/test/File.hpp"
+
+namespace Os {
+namespace Test {
+namespace File {
+
+//! Set up for the test ensures that the test can run at all
+//!
+void setUp(bool requires_io) {
+    if (requires_io) {
+        GTEST_SKIP() << "Cannot run tests requiring functional i/o";
+    }
+    Os::Stub::File::Test::StaticData::setNextStatus(Os::File::OP_OK);
+    Os::Stub::File::Test::StaticData::setPositionResult(0);
+}
+std::vector<std::shared_ptr<const std::string> > FILES;
+//! Tear down for the tests cleans up the test file used
+//!
+void tearDown() {}
+
+class StubsTester : public Tester {
+    //! Check if the test file exists.
+    //! \return true if it exists, false otherwise.
+    //!
+    bool exists(const std::string &filename) const override {
+        for (size_t i = 0; i < FILES.size(); i++) {
+            if (filename == *FILES.at(i)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    //! Get a filename, randomly if random is true, otherwise use a basic filename.
+    //! \param random: true if filename should be random, false if predictable
+    //! \return: filename to use for testing
+    //!
+    std::shared_ptr<const std::string> get_filename(bool random) const override {
+        std::shared_ptr<const std::string> filename = std::shared_ptr<const std::string>(new std::string("DOES-NOT-MATTER"), std::default_delete<std::string>());
+        FILES.push_back(filename);
+        return filename;
+    }
+
+    //! Posix tester is fully functional
+    //! \return true
+    //!
+    bool functional() const override {
+        return false;
+    }
+
+};
+
+std::unique_ptr<Os::Test::File::Tester> get_tester_implementation() {
+    return std::unique_ptr<Os::Test::File::Tester>(new Os::Test::File::StubsTester());
```

Comment:
See previous comment about std::unique_ptr.

Reply:
Same as above.  I wish this was supported as it is cleaner.